src/recipe/test/mocks/recipe.service.mock.ts

```diff
@@ -0,0 +1,22 @@
+export class RecipeServiceMock {
```

Comment:
you could make sure that RecipeServiceMock correctly implements RecipeService by adding some types.
```suggestion
export class RecipeServiceMock implements Required<RecipeService>{
```